numba/runtime/nrt.c

```diff
@@ -61,6 +61,14 @@ struct MemSys {
 /* The Memory System object */
 static NRT_MemSys TheMSys;
 
+
+typedef struct {
+    size_t allocated_size;
+    NRT_dtor_function dtor;
```

Comment:
Why not change the `NRT_dtor_function` signature to take `(ptr, size, info)` parameters? This would simplify things a lot.